tap-version-14-specification.md

```diff
@@ -0,0 +1,741 @@
+
+Lines written to standard output matching `/^(not)? ok\b/` must be
+interpreted as test lines. After a test line a block of lines starting
+with `/^ {2}---$/` and ending with `/^ {2}\.\.\.$/` will be interpreted as an
+inline YAML document providing extended diagnostic information about
+the preceding test.
+
+All other lines must not be considered test output.
+
+## TESTS LINES AND THE PLAN
+
+### The Version
+
+To indicate that this is TAP14 the first non-comment, non-whitespace
+line must be:
+
+```
+TAP version 14
+```
+
+A compliant TAP14 Harness SHOULD interpret TAP streams with `TAP
+version 13` according to TAP13 specification.
+
+TAP streams lacking a version number SHOULD be interpreted according to
+the TAP12 specification.
+
+In order to facilitate forward compatibility, a compliant TAP14
+Harness MAY interpret test data according to TAP14, regardless of
+stated version number.  For example, it may accept subtests in TAP12
+or TAP13 streams, even though these were not supported in those
+versions of TAP.
+
+Any stated version number less than 13 is an error.
+
+### The Plan
+
+The Plan tells how many tests will be run, or how many tests have run.
+It's a check that the test file hasn't stopped prematurely.
+
+The Plan MUST appear exactly once, EITHER:
+
+- the first line of TAP output after the Version, or
+- the last line of TAP output prior to the end of the TAP stream.
+
+The Plan matches the regular expression: `/^1\.\.(0|[1-9][0-9]*)( #.*)?$/`.
+
+The Plan specifies how many test points are expected in the test
+stream. For example,
+
+```
+1..10
+```
+
+means you plan on running 10 tests. This is a safeguard in case your
+test file dies silently in the middle of its run.
+
+In certain instances a test file may not know how many test points it
+will ultimately be running. In this case the Plan should be the last
+non-diagnostic line in the output.
+
+A test set missing a Plan MUST be interpreted as a failure by the
+harness.
+
+### The test line
+
+The core of TAP is the test line. A test file prints one test line
+test point executed. There must be at least one test line in TAP
+output. Each test line comprises the following elements:
+
+- ok or not ok
+
+This tells whether the test point passed or failed. It must be at the
+beginning of the line. /^not ok/ indicates a failed test point. /^ok/
+is a successful test point. This is the only mandatory part of the
+line.  Note that unlike the Directives below, ok and not ok are
+case-sensitive.
+
+- Test number
+
+TAP expects the ok or not ok to be followed by a test point number,
+separated by 1 or more space characters from the `/^(not )?ok/`.
+
+For example:
+
+```
+1..3
+ok 1
+not ok 2
+ok 3
+```
+
+If there is no number the harness must maintain its own counter until
+the script supplies test numbers again. So the following test output
+
+```
+1..6
+not ok
+ok
+not ok
+ok
+ok
+```
+
+has five tests. The sixth is missing.  For example, in Perl,
```

Comment:
Can this be made independent to language or sections like this clearly demarcated?

The W3C mark these as 'non normative'